pkg/dockerauth/multi.go

```diff
@@ -0,0 +1,32 @@
+package dockerauth
+
+import (
+	"fmt"
+
+	"github.com/fsouza/go-dockerclient"
+)
+
+type multiAuthProvider struct {
+	providers []AuthProvider
+}
+
+func NewMultiAuthProvider(providers ...AuthProvider) *multiAuthProvider {
+	return &multiAuthProvider{
+		providers: providers,
+	}
+}
+
+func (p *multiAuthProvider) GetAuthConfiguration(registry string) (docker.AuthConfiguration, error) {
+	for _, provider := range p.providers {
+		authConf, err := provider.GetAuthConfiguration(registry)
+		if err == nil {
```

Comment:
Another option, instead of using an error, could be to return a `*docker.AuthConfiguration` and give semantic meaning to nil.